src/pybind/rbd/rbd.pyx

```diff
@@ -2284,8 +2308,8 @@ cdef class PoolMetadataIterator(object):
                         c_vals[:vals_size].split(b'\0') if val]
             if len(keys) > 0:
                 free(self.last_read)
-                self.last_read = strdup(keys[-1])
-            self.next_chunk = zip(keys, vals)
+                self.last_read = strdup(cstr(keys[-1], 'last_read'))
```

Comment:
centos7 [1] does not like this:
```
Error compiling Cython file:
------------------------------------------------------------
...
                        c_keys[:keys_size].split(b'\0') if key]
            vals = [decode_cstr(val) for val in
                        c_vals[:vals_size].split(b'\0') if val]
            if len(keys) > 0:
                free(self.last_read)
                self.last_read = strdup(cstr(keys[-1], 'last_read'))
                                           ^
------------------------------------------------------------

rbd.pyx:2401:44: Obtaining 'char const *' from temporary Python value
```
[1] https://shaman.ceph.com/builds/ceph/wip-mgolub-testing/dc624d0b40a2d81426de469aa39fc3f3b5902dee/default/130675/